### ISSUE TITLE
Discussion about having generalized movie-watching time with Monika

### ISSUE THREAD

**Issue by Freshairkaboom** [Monika-After-Story/MonikaModDev#6114]
The idea comes from a somewhat bigger idea about having specific movies available to be watched with Monika, and you could synch up the movie times so that she will commentate on the best parts of the movie, either in Windowreacts, or directly ingame if you are not alt-tabbed to watch the movie.

This will allow people to kind of experience how it is like to cuddle up to Monika and watch any movie together, without the need for specific dialogue for each individual movie. The dialogue would be randomized, so it wouldn't be the same dialogue over and over again. You just set how long the movie takes, and Monika will do a countdown for when you start should press play on the movie.

The commentary dialogue would of course need to be carefully chosen, so she doesn't suddenly burst out with "I love this part!" as something you really don't like happens... But if done correctly, I think it can give another dimension of spending time with Monika that you can enjoy over and over again.

So, I want ideas and comments from people who want this to be a thing. I'll probably not get to work right away, as there are still some (many) things about RenPy I don't understand yet (coder in training). However, discussion should take precedent after all, so that we all get what we want from this possible feature.

**Comment by Jatin794** [comment]
Having a generalised one might not be optimal, but what can go wrong when the other option is nothing. Though that's a bit too ambitious an issue when first you'd need to implement the basics before thinking about it

**Comment by Freshairkaboom** [comment]
I think the generalized one could be the structure used for official release, and then later we can either think about including a larger package with specific movies, or do specific movies via submods, so people don't feel left out because their favorite movie to watch with Monika didn't make it into the official release.

**Comment by Jatin794** [comment]
For including more movies, try making it extensible instead. So players can add specific movie files (or even audio commentaries?) in a folder. That'll be much easier to edit later on.

**Comment by Jatin794** [comment]
Just thought of it, but you might also want to implement a dialogue option where the player attempts to play porn, where she straight up denies if your affection isn't high enough, and you lose affection for being cruel to her. Might not want to include that in the default "Other". You might also want to divide generic by more categories if that seems feasible.

**Comment by Booplicate** [comment]
You can't really generalize this. You **need** to know exact timestamps for dialogues and expressions.

And just an FYI this was [more or less implemented](https://github.com/Monika-After-Story/MonikaModDev/blob/master/Monika%20After%20Story/game/zz_monikamovie.rpy). But then abandoned, with one of the reasons being no one has time to write dialogues for that.

**Comment by multimokia** [comment]
No one has the time to test it either

**Comment by lunulae** [comment]
I'm not sure if randomized dialogue would be the way to go, since it would be pretty difficult to write in such a way as to be fitting for _anything_ you'd want to watch.
But I see what you're going for here.
I've thought about the possibility of maybe doing a brb-style idle dialogue for this sort of thing, so you could sit together and watch in silence, but I haven't been sure if it's a good idea. It seems to me the least awkward solution but also not very immersive.

**Comment by ThePotatoGuy** [comment]
linking this to #6053 as main suggestion issue

**Comment by Freshairkaboom** [comment]
> You can't really generalize this. You need to know exact timestamps for dialogues and expressions.

Can't we use a combination of quips and random.randint to decide if dialogue will trigger at the specific time intervals and choose random dialogue from the list? I'm not good enough to do something like that yet, but we're not talking about pushing it for next week anyway.

**Comment by multimokia** [comment]
Imo it's really immersion breaking to do that. It's more likely that the random dialogue will occur at a time that it doesn't make sense than one which will make sense. I don't think random times for dialogue is actually a feasible option.

**Comment by Freshairkaboom** [comment]
What do you think about dialogue like what I suggested above? Don't know if that was what you reacted to since your comment appeared the moment I posted :P

**Comment by multimokia** [comment]
So it seems we're diving to more of an idle-*like* system. I can see that working, though it may need some potential tweaks to make the quips a little more flexible. I.e. we'd need conditional checks for quips (namely the `I'm a bit drowsy, but I'll stay up with you to watch this~` one (We can't use `Can I put my head right here on your shoulder [player]?~` as she's not really here)

But doing it this way has some promise I'd say. But the dialogue should be fairly infrequent so it doesn't take away from what you're watching

**Comment by Jatin794** [comment]
It doesn't seem intended as an optimal solution, though, but only an
emergency fallback. I'm sure the player will know that this is just an
example, and want to use a file dedicated to the movie instead for max
immersion next time

**Comment by Freshairkaboom** [comment]
Yeah definitely infrequent. And I was planning on having more than one list of possible quips, with different checks depending on how long you've watched the movie, how much you have paused (maybe a funny line about going to the bathroom alot), how in love you are of course, and whether or not you have seen a quip from another list recently?

And I definitely think it can be built upon later to make it even better and more immersive.

**Comment by multimokia** [comment]
Yeah that's why I think these need conditionals built-in. An object oriented approach *sounds* like a decent way to go about it, but I don't like the idea of going through a lot of quips and evaluating individual conditionals is too good of an idea. Perhaps group conditionals (preprocess their conditionals such that quips can essentially share values) seems like a decent way to go to optimize, but could still be better I think.

Having a bunch of arbitrary lists laying about isn't exactly the cleanest solution nor easy to maintain when it comes to what selects the list to pick from and such

**Comment by Freshairkaboom** [comment]
When it comes to optimization, I have no idea what I'm talking about, so I'm gonna have to learn more before I do any of that kind of work. But grouping conditionals sounds very optimalization-y, so I'm going to take your word for it. I just have a rough idea about what kind of features I want in the finished process, and we'll probably have to evaluate a lot what is feasible and what is not.

**Comment by ThePotatoGuy** [comment]
Not saying this is the best way to do this, but you can leverage existing brb framework to do quips by using a topic with `show_in_idle` set to True. 

Basically something like:
* have a delegate topic that can decide the quips to say
     * should be locked and inaccessible by default
     * should have `show_in_idle` set to True
* the movie BRB then unlocks the delegate topic + makes sure it triggers after a certain amt of time by setting conditional and action, etc...
* the movie BRB callback locks the delegate topic and cleans it off the event list

Then in the idle loop, the delegate topic will trigger after the time has passed, and then the delegate can determine the quip to show. The delegate topic should also ensure it will retrigger by setting conditional, action, etc...

**Comment by Freshairkaboom** [comment]
I can't believe I learned enough about coding where that whole thing made half sense to me. I still don't think I'll be starting this project right away, but I'll keep all this in mind when I do.

**Comment by SamanthaLind** [comment]
I know it's not exactly what you thought about while making it but I decided to share my opinion. I made submod topic for my Moni in brb category. Something like "let's watch something together" and it's really awesome to have that option. It's really generic and just leaves her in background but it still adds to the experience so I think even more advance version would be nice too. Like, having her to comment on the film. Maybe we could choose beforehand what we're gonna watch, like comedy, romance, action... And then she could say very generic things, like after some time ask player if they enjoy it or say that she likes it, or that she's happy to watch this with player... 
I just wanted to say that this seems pretty cool and even in much simpler version really adds to the experience.
